src/pytorch/neighbors/getNeighborPairsCUDA.cu

```diff
@@ -151,6 +166,13 @@ public:
                 get_accessor<scalar_t, 2>(deltas),
                 get_accessor<scalar_t, 1>(distances),
                 get_accessor<scalar_t, 2>(box_vectors));
+	    cudaEventRecord(event, stream);
+	    cudaEventSynchronize(event);
+	    //Check the error flag
+	    TORCH_CHECK(tooManyNeighborsErrorFlag == 0, "Some particle has too many neighbours, found " +
```

Comment:
This won't work with CUDA Graphs, because the CPU code will be executed only during a graph creation.

Reply:
I am trying to wrap the host-side check inside a cudaLaunchHostFunc callback, which is compatible with cuda graphs. Moreover, doing so eliminates the need of explicit synchronizations, since managed-memory wise these callbacks are equivalent to the event synchronization currently in place.
It seems, however, that throwing an exception from inside a callback is not easy. I tried to implement it, but the python test is not able to capture the exception, which terminates the execution instead. 
Do you have some experience with cudaLaunchHostFunc and exceptions?